pkg/async/pool/pool.go

```diff
@@ -166,7 +166,12 @@ func (p *Pool) run(ctx context.Context) {
 		prevSize, delta, size int
 		cancellations         = cancellations{}
 	)
-	for ctx.Err() == nil {
+
+	if ctx.Err() != nil {
+		return
+	}
+
+	for {
```

Comment:
I think it is still better to retain a `for ctx.Err() == nil { ... }` here because go does not guarantee that the switch statement will select the `<- ctx.Done()` clause immediately after a context is canceled -- it might pick up the retry etc.

Reply:
+1 to this

Reply:
Ah, thanks for the reminder on this.